internal/controllers/app_controller.go

```diff
@@ -173,6 +196,18 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App) reconci
 			message: err.Error(),
 		}
 	}
+
+	// check for canary deployment
+	if app.Spec.Canary.IsActiveCanary {
+		app.DoCanary(metav1.NewTime(r.Now()))
+		if err := r.Update(ctx, app); err != nil {
```

Comment:
this Update makes ketch commands that change app CRD unreliable because an app CRD can be changed by the controller and the commands.

Reply:
We need to update deployments for increasing/decreasing the `weight` under `deployment.routingSettings` . So we should update the deployment. I am not sure how it makes ketch commands unreliable when if fetches the realtime results from the controller. Please let me know if I am missing something.

```
spec:
  canary:
    nextSchedule: '2021-01-21T15:50:31Z'
    stepTimeInterval: 1h
    stepWeight: 1
    steps: 5
  deployments:
    - exposedPorts:
        - port: 80
          protocol: TCP
      image: docker.io/dockercloud/hello-world
      processes:
        - cmd:
            - /bin/sh
            - '-c'
            - /run.sh
          name: web
      routingSettings:
        weight: 99
      version: 7
```